lib/miq_expression.rb

```diff
@@ -1373,8 +1367,8 @@ def to_arel(exp, tz)
       # Only support for tags of the main model
       if exp[operator].key?("tag")
         tag = Tag.parse(exp[operator]["tag"])
-        ids = tag.model.find_tagged_with(:any => parsed_value, :ns => tag.namespace).pluck(:id)
-        tag.model.arel_attribute(:id).in(ids)
+        ids = tag.target.find_tagged_with(:any => parsed_value, :ns => tag.namespace).pluck(:id)
```

Comment:
I can change this to not pluck the id and become an inner query.
It would remove some arel, but would add a few lines to make that work smothly